crates/bevy_math/src/curve/mod.rs

```diff
@@ -968,6 +968,73 @@ where
     }
 }
 
+/// Convert a `start` and `end` point of some type into a [`Curve`], sampled by interpolating
+/// between the two boundary values and under the application of an easing function.
+pub fn easing_curve<T: Clone>(
+    domain: Interval,
+    start: T,
+    end: T,
+    easing_fn: impl Fn(f32) -> f32,
+) -> impl Curve<T>
+where
+    T: std::ops::Mul<f32, Output = T>
+        + std::ops::Add<T, Output = T>
+        + std::ops::Div<f32, Output = T>,
```

Comment:
Is this assortment of traits better than just using `VectorSpace`?

Reply:
Oh I didn't know about that. That's cool! Will use it even though it's a bit more restrictive (requires `Copy`). But I think this is fine since `Add` and `Mul` also kind of require `Copy` already.

Reply:
Yeah, it's true that it's more restrictive, but the design of `bevy_math` so far kind of favors it as the relevant bundle of traits in situations like this. (And it seems pretty natural in a situation like this, where the basic object is literally a line.)